auth_cas/controllers/main.py

```diff
@@ -0,0 +1,183 @@
+# -*- coding: utf-8 -*-
+# © 2016 Roméo Guillot Roméo Guillot (http://www.opensource-elanz.fr).
+# © 2016-2018 Elico Corp (https://www.elico-corp.com).
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+import urllib
+from openerp.addons.web.controllers.main import login_and_redirect
+from openid.cryptutil import randomString
+from openerp.exceptions import AccessDenied
+import urlparse
+
+from ..pycas import login
```

Comment:
We should not have `pycas` as an embed library, we should add a fixed and stable version pip install on Travis and add this as external dependency named on __opernerp__.py

See OCA guidelines https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#17external-dependencies

See project https://pypi.org/project/pycas/

See an example https://github.com/OCA/reporting-engine/blob/11.0/report_xlsx/__manifest__.py#L16